pandas/io/parsers.py

```diff
@@ -1791,20 +1791,10 @@ def __init__(self, f, **kwds):
         self.comment = kwds['comment']
         self._comment_lines = []
 
-        if isinstance(f, compat.string_types):
-            f = _get_handle(f, 'r', encoding=self.encoding,
-                            compression=self.compression,
-                            memory_map=self.memory_map)
-            self.handles.append(f)
-        elif self.compression:
-            f = _wrap_compressed(f, self.compression, self.encoding)
-            self.handles.append(f)
-        # in Python 3, convert BytesIO or fileobjects passed with an encoding
-        elif compat.PY3 and isinstance(f, compat.BytesIO):
-            from io import TextIOWrapper
-
-            f = TextIOWrapper(f, encoding=self.encoding)
-            self.handles.append(f)
+        f = _get_handle(f, 'r', encoding=self.encoding,
+                        compression=self.compression,
+                        memory_map=self.memory_map)
+        self.handles.append(f)
```

Comment:
@jorisvandenbossche I attempted a fix in 2f43c39e16b844a0f5552c407225387135d38e18, but it's not very elegant -- logic is repeated in ` _get_handle` and `PythonParser.__init__`.

What do you think is the best solution? What situation do we not end up opening a handle ourselves? Is it a problem that in PY3 we pass `path_or_buf` through `io.TextIOWrapper` before adding it to `handles`?